service/matching/version_rule_helpers.go

```diff
@@ -366,9 +366,10 @@ func checkAssignmentConditions(g *persistencespb.VersioningData, maxARs int, req
 
 // checkRedirectConditions checks for validity conditions that must be assessed by looking at the entire set of rules.
 // It returns an error if the new set of redirect rules don't meet the following requirements:
-// - No more rules than dynamicconfig.VersionRedirectRuleLimitPerQueue
-// - The DAG of redirect rules must not contain a cycle
-// - The DAG of redirect rules must not contain a chain of connected rules longer than dynamicconfig.VersionRedirectRuleChainLimitPerQueue
+//   - No more rules than dynamicconfig.VersionRedirectRuleLimitPerQueue
+//   - The DAG of redirect rules must not contain a cycle
+//   - The DAG of redirect rules must not contain a chain of connected rules longer than dynamicconfig.VersionRedirectRuleChainLimitPerQueue
+//     (Here, a "chain" counts the # of vertices, not the # of edges. So 3 ---> 4 ---> 5 has a chain length of 3)
```

Comment:
Q: what's the chain length of the following? 3 or 4?
```
3 --> 4 --> 5
2 ----^
```

Reply:
the "maxChainLength" of that would be 3.

`3 --> 4 --> 5` and `2--> 4 --> 5` would each be considered their own chain, both with length 3